lib/routes/user/home/home_page.dart

```diff
@@ -35,14 +38,19 @@ class Home extends StatefulWidget {
   final UserProfileBloc userProfileBloc;
   final ConnectPayBloc ctpBloc;
   final BackupBloc backupBloc;
+  final GlobalKey<NavigatorState> navigatorKey;
 
-  Home(this.accountBloc, this.invoiceBloc, this.userProfileBloc, this.ctpBloc, this.backupBloc) {
-    _minorActionsInvoice =
-    new List<DrawerItemConfig>.unmodifiable([
-      new DrawerItemConfig(
-          "/pay_invoice", "Pay Invoice", "src/icon/qr_scan.png", onItemSelected: (String name) async {
+  Home(this.accountBloc, this.invoiceBloc, this.userProfileBloc, this.ctpBloc, this.backupBloc, this.navigatorKey) {
+    _minorActionsInvoice = new List<DrawerItemConfig>.unmodifiable([
+      new DrawerItemConfig("/pay_invoice", "Pay Invoice", "src/icon/qr_scan.png", onItemSelected: (String name) async {
+        Map<PermissionGroup, PermissionStatus> permissions = await PermissionHandler().requestPermissions([PermissionGroup.camera]);
```

Comment:
Permission check is required because using a failed Future instead of knowing permissions beforehand would mean we'd show the barcode plugin momentarily, which causes a rendering issue.

[The fallback is added in case the permission is given or in unknown state but there's a failure.](https://github.com/breez/breezmobile/pull/146/commits/8b8bc0445a7d69feb6327ca26e27584433b7d0b9)